.circleci/config.yml

```diff
@@ -0,0 +1,69 @@
+---
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: debian:jessie
+
+    environment:
+      STACK_VERSION: 1.6.5
+      GHC_8_2_RESOLVER: nightly-2017-12-16
+
+    steps:
+      - checkout:
+          path: ~/stratosphere
+
+      - run:
+          name: Setup Environment Variables
+          command: |
+            echo 'export PATH="$HOME/.local/bin:$PATH"' >> $BASH_ENV
+
+      - run:
+          name: Install System Dependencies
+          command: |
+            apt-get update --assume-yes
+            apt-get install --assume-yes curl build-essential libgmp3-dev
+
+      - run:
+          name: Install Dependencies
+          # yamllint disable-line rule:line-length
+          command: |
+            mkdir -p ~/.local/bin
+            curl -L https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/stack-$STACK_VERSION-linux-x86_64.tar.gz \
+              | tar -xzv -C ~/.local/bin \
+              && ln -vsf ~/.local/bin/stack-*/stack ~/.local/bin/stack
+
+      - run:
+          name: Build
+          command: |
+            stack setup --resolver $GHC_8_2_RESOLVER
+            stack build --resolver $GHC_8_2_RESOLVER stratosphere --test \
+              --only-dependencies --no-run-tests --jobs=1 \
+              --flag stratosphere:-library-only
+            stack setup
+            stack build stratosphere --test --only-dependencies \
+              --no-run-tests --jobs=1 --flag stratosphere:-library-only
+
+      - run:
+          name: Test
+          command: |
+            stack build --resolver $GHC_8_2_RESOLVER stratosphere --pedantic \
+              --test --jobs=1 --flag stratosphere:-library-only
+            stack haddock --resolver $GHC_8_2_RESOLVER stratosphere \
+              --no-haddock-deps --jobs=1 --flag stratosphere:-library-only
+            stack build stratosphere --pedantic --test --jobs=1 \
+              --flag stratosphere:-library-only
+            stack haddock stratosphere --no-haddock-deps --jobs=1 \
+              --flag stratosphere:-library-only
+          no_output_timeout: 30m
+
+      - save_cache:
+          key: dependency-cache
```

Comment:
Hmm, this won't work. After the first build Circle will never re-upload the cache since the key doesn't change. Take a look at the [caching docs](https://circleci.com/docs/2.0/caching/) and also take a look at the `megarepo` to see how we generally cache Haskell builds.

I recommend just caching dependencies. No need to cache `.stack-work`.

Reply:
Will look closer - I was just following the migration guide and this looked correct!